python/src/scipp/plot/figure2d.py

```diff
@@ -0,0 +1,164 @@
+# Scipp imports
+from .. import config
+from .tools import get_mpl_axes
+
+# Other imports
+import numpy as np
+import ipywidgets as ipw
+import matplotlib.pyplot as plt
+import warnings
+import io
+# from copy import deepcopy
+
+
+class PlotFigure2d:
+    def __init__(self,
+                 ax=None,
+                 cax=None,
+                 figsize=None,
+                 aspect=None,
+                 cmap=None,
+                 norm=None,
+                 title=None,
+                 cbar=None,
+                 unit=None,
+                 log=None,
+                 vmin=None,
+                 vmax=None,
+                 color=None,
+                 logx=False,
+                 logy=False,
+                 mask_cmap=None,
+                 masks=None,
+                 resolution=None):
+
+        # Get matplotlib figure and axes
+        self.fig, self.ax, self.cax, self.own_axes = get_mpl_axes(
+            ax=ax, cax=cax, figsize=figsize)
```

Comment:
Is there a relevant amount of code shared with `PlotFigure1d` so a common base class (for a generic MPL figure) would make sense? I do see a number of duplicate lines.

Reply:
Good idea. Will do this